frontend/src/pages/hardwareProfiles/nodeResource/NodeResourceTable.tsx

```diff
@@ -1,30 +1,72 @@
 import React from 'react';
-import { Td, Tr } from '@patternfly/react-table';
 import { Table } from '~/components/table';
 import { Identifier } from '~/types';
-import { nodeResourceColumns } from '~/pages/hardwareProfiles/const';
+import { nodeResourceColumns } from './const';
+import NodeResourceTableRow from './NodeResourceTableRow';
+import ManageNodeResourceModal from './ManageNodeResourceModal';
 
 type NodeResourceTableProps = {
   nodeResources: Identifier[];
+  onUpdate?: (identifiers: Identifier[]) => void;
+  viewOnly?: boolean;
 };
 
-const NodeResourceTable: React.FC<NodeResourceTableProps> = ({ nodeResources }) => (
-  <Table
-    variant="compact"
-    data-testid="hardware-profile-node-resources-table"
-    id="hardware-profile-node-resources-table"
-    data={nodeResources}
-    columns={nodeResourceColumns}
-    rowRenderer={(cr, index) => (
-      <Tr key={index}>
-        <Td dataLabel="Resource label">{cr.displayName}</Td>
-        <Td dataLabel="Resource identifier">{cr.identifier}</Td>
-        <Td dataLabel="Default">{cr.defaultCount}</Td>
-        <Td dataLabel="Minimum allowed">{cr.minCount}</Td>
-        <Td dataLabel="Maximum allowed">{cr.maxCount}</Td>
-      </Tr>
-    )}
-  />
-);
+const NodeResourceTable: React.FC<NodeResourceTableProps> = ({
+  nodeResources,
+  onUpdate,
+  viewOnly,
+}) => {
+  const [editIdentifier, setEditIdentifier] = React.useState<Identifier | undefined>();
+  const [currentIndex, setCurrentIndex] = React.useState<number | undefined>();
+
+  return (
+    <>
+      <Table
```

Comment:
You will need to use `TableBase` instead of `Table` here. Otherwise, It will automatically sort on the first column, which is `Resource label` even if we set `sortable: false` here.